### PR TITLE
Fix use of `OPENVINO_ASSERT`

### PR DESCRIPTION
**Pull request by vurusovs**
### Details:
 - `OPENVINO_ASSERT` takes boolean expression as first argument. In order to explicitly throw AssertFailure, need to specify `false`, non-empty string will be treated as `true`

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 1]
```diff
@@ -2002,7 +2002,7 @@ bool primitive_inst::is_valid_fusion() const {
         if (fd.is_type<eltwise>() || fd.is_type<activation>()) {
             fused_eltwise_prims.push_back(fd);
         } else {
-            OPENVINO_ASSERT("[GPU] Unsupported fused operation in dynamic shape : ", fd.desc->id);
+            OPENVINO_ASSERT(false, "[GPU] Unsupported fused operation in dynamic shape : ", fd.desc->id);
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:2005]: Maybe it's better to replace it with `OPENVINO_THROW` instead?

Comment (vurusovs) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp]: Done, thank you!

File: src/plugins/intel_gpu/src/graph/reshape.cpp [hunk 2]
```diff
@@ -24,7 +24,7 @@ layout reshape_inst::calc_output_layout(reshape_node const& node, kernel_impl_pa
         if (desc->output_partial_shape.size() != 0) {
             return layout{desc->output_partial_shape, input_layout.data_type, input_layout.format};
         } else {
-            OPENVINO_ASSERT("[GPU] Output shape is not provided");
+            OPENVINO_THROW("[GPU] Output shape is not provided");
```
Comment (vurusovs) [inline src/plugins/intel_gpu/src/graph/reshape.cpp:27]: It causes test failure in precommit

Comment (vurusovs) [reply src/plugins/intel_gpu/src/graph/reshape.cpp:27]: Should be fixed

**Review by p-durandin — APPROVED** [review]

**Pull request merged**